pkg/metricprovider/interface.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+// Package metricprovider features interfaces and other components that can be used to build
+// a Fleet metric provider.
+package metricprovider
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+)
+
+// MetricCollectionResponse is returned by a Fleet metric provider to report metrics and
+// metric collection status.
+type MetricCollectionResponse struct {
+	// Metrics is an array of metrics and their values. The key should be the name of the
+	// metric, which is a Kubernetes label name; the value is the metric data.
+	Metrics map[string]float64
+	// Resources is a group of resources, described by their allocatable capacity and
+	// available capacity.
+	Resources clusterv1beta1.ResourceUsage
```

Comment:
I am not sure if I get how to combine these two together.

Reply:
and how can this be translated to internalMemberCluster fields

Reply:
The two are not combined; non-resource and resource metrics are kept separately. The metrics returned here are put in the `metrics` field in the `InternalMemberCluster` status (with the value being serialized as a string); the resources go to the `resourceUsage` field in the status.